src/app/core/common-components/entity-select/entity-select.component.ts

```diff
@@ -139,86 +128,87 @@ export class EntitySelectComponent<
-      this.includeInactive || entity.isActive;
-    const includeSelected = (entity: E) =>
-      asArray(this.form.value).includes(entity.getId());
-
-    const newAvailableEntities = this.allEntities.filter(
-      (e) => includeInactive(e) || includeSelected(e),
-    );
+  private entitiesForType: Resource<E[]> = resource({
+    defaultValue: [],
+    params: () => ({
+      entityTypes: this.entityType(),
+    }),
+    loader: async ({ params }) => {
+      if (params.entityTypes.length === 0) return [];
+
+      const entities: E[] = [];
+      for (const type of params.entityTypes) {
+        entities.push(...(await this.entityMapperService.loadType<E>(type)));
+      }
 
-    await this.alignAvailableAndSelectedEntities(newAvailableEntities);
-
-    this.availableOptions.next(newAvailableEntities);
-    this.recalculateMatchingInactive();
-  }
+      return entities
+        .filter((e) => this.additionalFilter(e))
+        .sort((a, b) => a.toString().localeCompare(b.toString()));
+    },
+  });
 
   /**
-   * Edit form value (currently selected) and the given available Entities to be consistent:
-   * Entities that do not exist should be removed from the form value
-   * and availableEntities should contain all selected entities, even from other types.
-   * @private
+   * true when this is loading and false when it's ready.
+   * This subject's state reflects the actual loading resp. the 'readiness'-
+   * state of this component. Will trigger once loading is done
    */
-  private async alignAvailableAndSelectedEntities(availableEntities: E[]) {
-    if (this.form?.value === null || this.form?.value === undefined) {
-      return;
-    }
-
-    let updatedValue: T = this.form.value;
+  loading: Signal<boolean> = computed(() => this.entitiesForType.isLoading());
 
-    for (const id of asArray(this.form.value)) {
-      // Skip null, undefined, or empty string values
-      if (id === null || id === undefined || id === "") {
-        continue;
-      }
+  /**
+   * The currently selected values (IDs) of the form control.
+   */
+  values: Signal<string[]> = toSignal(
+    toObservable(this.form)
+      .pipe(switchMap((form) => form.valueChanges))
+      .pipe(map(asArray)),
+    { initialValue: [] },
+  );
+
+  private availableOptionsResource: Resource<E[]> = resource({
+    defaultValue: [],
+    params: () => ({
+      allEntities: this.entitiesForType.value(),
+      values: this.values(),
+      includeInactive: this.includeInactive(),
+    }),
+    loader: async ({ params }) => {
+      const includeSelected = (entity: E) =>
+        asArray(params.values).includes(entity.getId());
+
+      const availableEntities = params.allEntities.filter(
+        (e) => params.includeInactive || e.isActive || includeSelected(e),
+      );
 
-      if (availableEntities.find((e) => id === e.getId())) {
-        // already available, nothing to do
-        continue;
+      if (params.values !== null && params.values !== undefined) {
```

Comment:
with its initialValue and map(asArray), could we assume that `values` is never undefined?